privilege/conn/conn.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2023-2023 PingCAP Xingchen (Beijing) Technology Co., Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conn
+
+import "context"
+
+// AuthConn is an interface to help auth plugin to communicate with the client.
+type AuthConn interface {
```

Comment:
It is a good habit to create a new interface in the new package. It can improve build performance with bazel. 👍